### PR TITLE
check ocp311 version

### PR DESCRIPTION
**Pull request by zhiweiyin318**
issue: https://github.com/open-cluster-management/backlog/issues/6965

### CODE REVIEW

File: pkg/klusterlet/clusterinfo/clusterinfo_controller.go [hunk 1]
```diff
@@ -471,6 +472,11 @@ func (r *ClusterInfoReconciler) getOCPDistributionInfo() (clusterv1beta1.OCPDist
 	ocpDistributionInfo := clusterv1beta1.OCPDistributionInfo{}
 	obj, err := r.ManagedClusterDynamicClient.Resource(ocpVersionGVR).Get(context.TODO(), "version", metav1.GetOptions{})
 	if err != nil {
```
Comment (qiujian16) [inline pkg/klusterlet/clusterinfo/clusterinfo_controller.go:474]: should we have a more careful err check? And unit test please.
on vanilla kube, the version should not set. On ocp, it should set.

Comment (zhiweiyin318) [reply pkg/klusterlet/clusterinfo/clusterinfo_controller.go:474]: `getOCPDistributionInfo` is called only when the cluster is OCP.  so I think `NotFound` is enough for check OCP 3.11 here.
UT is added.

**Pull request merged**